pkg/gateway/filter/handler/grpc/handler.go

```diff
@@ -0,0 +1,587 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package grpc provides grpc server logic
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kpango/fuid"
+	"github.com/vdaas/vald/apis/grpc/gateway/filter"
+	"github.com/vdaas/vald/apis/grpc/gateway/vald"
+	"github.com/vdaas/vald/apis/grpc/payload"
+	client "github.com/vdaas/vald/internal/client/gateway/vald"
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/info"
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/net/grpc"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+	"github.com/vdaas/vald/internal/observability/trace"
+	"github.com/vdaas/vald/internal/safety"
+	"github.com/vdaas/vald/pkg/gateway/filter/service"
+	"github.com/vdaas/vald/pkg/gateway/tool/location"
+)
+
+type Server interface {
+	vald.ValdServer
+	filter.FilterServer
+}
+
+type server struct {
+	eg                errgroup.Group
+	filter            service.Filter
+	gateway           client.Client
+	copts             []grpc.CallOption
+	streamConcurrency int
```

Comment:
there is no `metadata` in server struct.
please confirm it for all service handler file 🙏

Reply:
filter gateway is another task please ignore them.